### PR TITLE
Add finalizer subcommand

### PR DESCRIPTION
**Pull request by Yuni-sa**
Iterates over all types of namespaced resources in the cluster (Including Custom Resources) check if they are in a terminating state by looking if they have a finalizer and a deletion timestamp, deletion of those resources is also supported by patching out the finalizer.

part of #117 

this command is not added to the "all" command for future plans implementing a separate function for the supported resources after #167

### CODE REVIEW

**Comment by yonahd** [comment]
@Yuni-sa is this ready for review?

File: pkg/kor/finalizers.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to process resources waiting for finalizers: %v\n", err)
+	}
+
+	allDiffs := make(map[string][]string)
+
+	for namespace, resourceType := range pendingDeletionDiffs {
+		if slices.Contains(namespaces, namespace) {
+			for gvr, resourceDiff := range resourceType {
```
Comment (yonahd) [inline pkg/kor/finalizers.go:100]: What is gvr?

**Review by yonahd — COMMENTED** [review]
Overall it looks good.
Can we support this in multi and all and merge the namespaces?
Maybe we can do it that way and then put finalizers as include finalizers flag or something like that?

WDYT?

File: pkg/kor/finalizers.go (hunk 1 above)
Comment (Yuni-sa) [reply pkg/kor/finalizers.go:100]: schema of group version resource, resource stands for the name of the resource in the same format of the api endpoint which is lowercase plural.
for example: Kind- `Pod`, Resource-`pods`

**Comment by Yuni-sa** [comment]
> Overall it looks good. Can we support this in multi and all and merge the namespaces? Maybe we can do it that way and then put finalizers as include finalizers flag or something like that?
> 
> WDYT?

Not sure if there is a good implementation of this without #167 , after reasoning will be added I will create a new pr that will include it in all and multi.

if we add this now it would probably all be deleted/changed in the next pr, I think the current implementation of using `kor finalizers` which also supports all of the filters options is fine for now.

**Comment by yonahd** [comment]
> > Overall it looks good. Can we support this in multi and all and merge the namespaces? Maybe we can do it that way and then put finalizers as include finalizers flag or something like that?
> > WDYT?
> 
> Not sure if there is a good implementation of this without #167 , after reasoning will be added I will create a new pr that will include it in all and multi.
> 
> if we add this now it would probably all be deleted/changed in the next pr, I think the current implementation of using `kor finalizers` which also supports all of the filters options is fine for now.

Ok.
Can you just add the command in the Readme and then we can merge it

**Review by yonahd — APPROVED** [review]
LGTM
Thanks!

**Pull request merged**